contrib/helm/harbor/README.md

```diff
@@ -4,24 +4,75 @@
 
 ## Introduction
 
-This is an experimental monolithic chart that installs and configures VMWare Harbor and its dependencies. The initial implementation of this includes all of the components required to run Harbor. As upstream harbor becomes more cloud native we will be able to break apart the monolith and utitlize helm dependencies.
+This chart installs and configures Harbor.
 
 ## Prerequisites
 
-- Kubernetes 1.7+ with Beta APIs enabled
+- Kubernetes cluster 1.8+ with Beta APIs enabled
+- Kubernetes Ingress Controller is enabled
+- kubectl CLI 1.8+
 - PV provisioner support in the underlying infrastructure
 
+## Setup a Kubernetes cluster
+
+You can use any tools to setup a K8s cluster.
+In this guide, we use [minikube](https://github.com/kubernetes/minikube) to setup a K8s cluster as the dev/test env.
+
+```bash
+# Start minikube
+minikube start --vm-driver=none
+# Enable Ingress Controller
+minikube addons enable ingress
+```
+
 ## Installing the Chart
 
-To install the chart with the release name `my-release`:
+First install [Helm CLI](https://github.com/kubernetes/helm#install), then initialize Helm.
+```bash
+helm init --canary-image
+```
+
+Download Harbor helm chart code.
+
+```bash
+git clone https://github.com/vmware/harbor/tree/master/contrib/helm/harbor helm-harbor
+cd helm-harbor
+```
+
+### Insecure Registry Mode
+
+If setting Harbor Registry as insecure-registries for docker,
+you don't need to generate Root CA and SSL certificate for the Harbor ingress controller.
+
+Install the Harbor helm chart with a release name `my-release`:
+
+```bash
+helm install . --debug --name my-release --set externalDomain=harbor.my.domain,insecureRegistry=true
+```
+
+**Make sure** `harbor.my.domain` resolves to the K8s Ingress Controller IP on the machines where you run docker or access Harbor UI.
+You can add `harbor.my.domain` and IP mapping in the DNS server, or in /etc/hosts, or use the FQDN `harbor.<IP>.xip.io`.
+
+Then add `"insecure-registries": ["harbor.my.domain"]` in the docker daemon config file and restart docker service.
+
+### Secure Registry Mode
+
+By default this chart will generate a root CA and SSL certificate for your Harbor.
+You can also use your own CA signed certificate:
+
+open values.yaml, set the value of 'externalDomain' to your Harbor FQDN, and
+set value of 'tlsCrt', 'tlsKey', 'caCrt'. The common name of the certificate must match your Harbor FQDN.
+
+Install the Harbor helm chart with a release name `my-release`:
 
 ```bash
-$ git clone https://github.com/vmware/harbor.git
-$ cd harbor/contrib/helm/harbor
-$ helm install --name my-release incubator/harbor
+helm install . --debug --name my-release --set externalDomain=harbor.my.domain
```

Comment:
I don't like removing the `$` from in front of the commands.  It's used to tell people that's the bit they need to run, and allows us to show the output of the commands.   It also discourages people from copy/pasting blocks of commands directly to a terminal which is a bad practice.

Reply:
I'd prefer not adding '$' for a command.

Reply:
without putting some sort of marker to show the command prompt like `$`, how does the user know what is input vs output ?  Currently our instructions are fairly simple and only show the output, but we may at some stage want to show the full output of a command ( example the `helm install` output of the NOTES.txt ).

Reply:
I understand your point.